File: codegen/results/v002/qwen3.6_35b_think/20260418_180142/typescript/005_unit_test_writer/solution.ts
function calculateDiscount(price: number, quantity: number): number {
    if (quantity < 10) return price * quantity;
    else if (quantity < 50) return price * quantity * 0.9;
    else return price * quantity * 0.75;
}

interface TestCase {
    description: string;
    price: number;
    quantity: number;
    expected: number;
}

const testCases: TestCase[] = [
    { description: "boundary: quantity 9", price: 100, quantity: 9, expected: 900 },
    { description: "boundary: quantity 10", price: 100, quantity: 10, expected: 900 },
    { description: "boundary: quantity 49", price: 100, quantity: 49, expected: 4410 },
    { description: "boundary: quantity 50", price: 100, quantity: 50, expected: 3750 }
];

const EPSILON = 1e-9;

for (const tc of testCases) {
    const result = calculateDiscount(tc.price, tc.quantity);
    const pass = Math.abs(result - tc.expected) < EPSILON;
    
    if (pass) {
        console.log(`PASS: ${tc.description}`);
    } else {
        console.log(`FAIL: ${tc.description} (expected: ${tc.expected}, got: ${result})`);
    }
}

File: codegen/results/v002/qwen3.6_35b_think/20260418_180142/go/005_unit_test_writer/solution.go
package main

import (
	"fmt"
	"math"
)

func calculateDiscount(price float64, quantity int) float64 {
	if quantity < 10 {
		return price * float64(quantity)
	} else if quantity < 50 {
		return price * float64(quantity) * 0.9
	}
	return price * float64(quantity) * 0.75
}

func main() {
	testCases := []struct {
		desc     string
		price    float64
		quantity int
		expected float64
	}{
		{"Quantity 9 (boundary < 10)", 100, 9, 900.0},
		{"Quantity 10 (boundary >= 10)", 100, 10, 900.0},
		{"Quantity 49 (boundary < 50)", 100, 49, 4410.0},
		{"Quantity 50 (boundary >= 50)", 100, 50, 3750.0},
	}

	for _, tc := range testCases {
		result := calculateDiscount(tc.price, tc.quantity)
		if math.Abs(tc.expected-result) < 1e-9 {
			fmt.Printf("PASS: %s\n", tc.desc)
		} else {
			fmt.Printf("FAIL: %s (expected: %v, got: %v)\n", tc.desc, tc.expected, result)
		}
	}
}